dowhy/causal_refuters/linear_sensitivity_analysis.py

```diff
@@ -0,0 +1,435 @@
+        :param r2tu_w: hypothetical partial R^2 of confounder with treatment(x - axis)
+        :param r2yu_tw: hypothetical partial R^2 of confounder with outcome(y - axis)
+
+        :returns:
+        contour_values : values of contour lines for the plot
+        critical_t : threshold point 
+        t_bounds : t-value for unobserved confounders (bias adjusted t values)
+        """
+
+        t_alpha_df_1 = t.ppf(self.significance_level / 2 , self.degree_of_freedom - 1) # t-value threshold with alpha significance level and dof-1 degrees of freedom
+        critical_t = abs(t_alpha_df_1) * np.sign(self.t_stats)
+
+        contour_values = []
+        for x in r2tu_w:
+            contour = []
+            for y in r2yu_tw:
+                benchmarking_results = self.compute_bias_adjusted(r2tu_w = x, r2yu_tw = y)
+                t_value = benchmarking_results['bias_adjusted_t']
+                contour.append(t_value)
+            contour_values.append(contour)
+
+        t_bounds = self.bias_adjusted_t
+        return contour_values, critical_t, t_bounds
+
+    
+    def plot(self, sensitivity_variable = 'estimate', critical_estimate = 0, critical_t = 2, contour_color = "blue", threshold_color = "red", x_limit = 0.8, y_limit = 0.8 ):
+        """
+        The horizontal axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the treatment
+        The vertical axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the outcome.
+        The contour levels represent adjusted t-values or estimates for unobserved confounders with these hypothetical partialR2 values.
+        We also plot bounds on the partial R^2 of the unobserved confounders obtained from observed covariates
+
+        :param sensitivity_variable: "estimate" or "t-value"
+        :param critical_estimate: threshold line to represent the contour corresponding to estimate, (default = 0)
+        :param critical_t: threshold line to represent the contour corresponding to t-value, (default = 2 - usual approx value for 95% CI)
+        :param x_limit: plot's maximum x_axis value
+        :param y_limit: plot's minimum y_axis value
+        :param contour_color: color of contour line(default = blue)
+        :param threshold_color: color of threshold line(default = red)
+        """
+
+        #Plotting the contour plot
+        fig, ax = plt.subplots(1, 1, figsize = (7,7))
+        ax.set_title("Sensitivity contour plot of %s"  %sensitivity_variable)
+        ax.set_xlabel("Partial R^2 of confounder with treatment")
+        ax.set_ylabel("Partial R^2 of confounder with outcome")
+
+        for i in range(len(self.r2tu_w)):
+            x = self.r2tu_w[i]
+            y = self.r2yu_tw[i]
+            if(x > 0.8 or y > 0.8):
+                x_limit = 0.95
+                y_limit = 0.95
+                break
+
+        r2tu_w = np.arange(0.0, x_limit, x_limit / 200)

```

Comment:
why 200?

Reply:
as we increase the number of points, the graph becomes more continuous but the computation time increases therefore after trying several values - 10, 50, 100, 400... I set it to 200. But we can also give it in param list

Reply:
got it. best to provide it as a param of the method. Is it the number of points per contour line? if so, can call it num_points_per_contour and set default as 200.